### PR TITLE
Icons renaming

### PR DESCRIPTION
**Pull request by vvalentin-lucca**
## Description

Icons updated with a completely revised naming system.

-----

- Icons updated with a completely revised naming system. (But remain accessible by removing the associated `display: none`).
- All LF icons adopt the [new nomenclature](https://www.figma.com/file/Qf2Y5sdvAKDcvHPaMtwCtAPm/Lucca-icons?type=design&node-id=38-814&mode=design&t=SSA5WvFXb8QuD11r-0).
-  **Breaking** for the star icon, which will change from a full shape to an empty one if nothing is modified. (You need to change the call to `star_filled` if you don't want any visual change).
- IE support also stopped.
- [The dedicated Icomoon project](https://icomoon.io/app/#/projects) is called “lucca icons (renaming)”.

-----

Capture d’écran 2023-07-21 à 17 09 31

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2170/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/4618/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2170/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/4662/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2170/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/4848/ui-diff)

File: packages/scss/src/components/tableSorted/states.scss [hunk 1]
```diff
@@ -5,7 +5,7 @@
 	color: var(--palettes-grey-900);
 
 	&::after {
-		@include icon.generate('arrow_full_north');
+		@include icon.generate('arrow_north');
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/tableSorted/states.scss:8]:
```suggestion
		@include icon.generate('arrow_top');
```

File: packages/scss/src/components/tableSorted/states.scss [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
 @mixin sortedLegacyAlignRight {
 	&::before {
-		@include icon.generate('arrow_full_north');
+		@include icon.generate('arrow_north');
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/tableSorted/states.scss:23]:
```suggestion
		@include icon.generate('arrow_top');
```

File: packages/scss/src/components/tableSorted/states.scss [hunk 3]
```diff
@@ -41,31 +41,31 @@
 	@include sortedLegacy;
 
 	&::after {
-		@include icon.generate('arrow_full_south');
+		@include icon.generate('arrow_north');
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/tableSorted/states.scss:44]:
```suggestion
		@include icon.generate('arrow_bottom');
```

File: packages/scss/src/components/tableSorted/states.scss [hunk 4]
```diff
@@ -41,31 +41,31 @@
...
 	}
 }
 
 @mixin ascendingLegacy {
 	@include sortedLegacy;
 
 	&::after {
-		@include icon.generate('arrow_full_north');
+		@include icon.generate('arrow_north');
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/tableSorted/states.scss:52]:
```suggestion
		@include icon.generate('arrow_top');
```

File: packages/scss/src/components/tableSorted/states.scss [hunk 5]
```diff
@@ -41,31 +41,31 @@
...
 	}
 }
 
 @mixin descendingLegacyAlignRight {
 	@include sortedLegacyAlignRight;
 
 	&::after {
-		@include icon.generate('arrow_full_south');
+		@include icon.generate('arrow_north');
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/tableSorted/states.scss:60]:
```suggestion
		@include icon.generate('arrow_bottom');
```

File: packages/scss/src/components/tableSorted/states.scss [hunk 6]
```diff
@@ -41,31 +41,31 @@
...
 	}
 }
 
 @mixin ascendingLegacyAlignRight {
 	@include sortedLegacyAlignRight;
 
 	&::after {
-		@include icon.generate('arrow_full_north');
+		@include icon.generate('arrow_north');
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/tableSorted/states.scss:68]:
```suggestion
		@include icon.generate('arrow_top');
```

File: packages/scss/src/components/tableSorted/states.scss [hunk 7]
```diff
@@ -77,7 +77,7 @@
 		padding: var(--components-table-padding);
 
 		&::after {
-			@include icon.generate('arrow_full_north');
+			@include icon.generate('arrow_north');
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/tableSorted/states.scss:80]:
```suggestion
			@include icon.generate('arrow_top');
```

File: packages/scss/src/components/tableSorted/states.scss [hunk 8]
```diff
@@ -102,7 +102,7 @@
 		text-align: right;
 
 		&::before {
-			@include icon.generate('arrow_full_north');
+			@include icon.generate('arrow_north');
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/tableSorted/states.scss:105]:
```suggestion
			@include icon.generate('arrow_top');
```

File: packages/scss/src/components/tableSorted/states.scss [hunk 9]
```diff
@@ -131,7 +131,7 @@
 
 	.table-head-row-cell-sortableButton {
 		&::after {
-			@include icon.generate('arrow_full_south');
+			@include icon.generate('arrow_north');
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/tableSorted/states.scss:134]:
```suggestion
			@include icon.generate('arrow_bottom');
```

File: packages/scss/src/components/tableSorted/states.scss [hunk 10]
```diff
@@ -141,7 +141,7 @@
 
 	.table-head-row-cell-sortableButton {
 		&::after {
-			@include icon.generate('arrow_full_north');
+			@include icon.generate('arrow_north');
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/tableSorted/states.scss:144]:
```suggestion
			@include icon.generate('arrow_top');
```

File: packages/scss/src/components/tableSorted/states.scss [hunk 11]
```diff
@@ -151,7 +151,7 @@
 
 	.table-head-row-cell-sortableButton {
 		&::before {
-			@include icon.generate('arrow_full_south');
+			@include icon.generate('arrow_north');
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/tableSorted/states.scss:154]:
```suggestion
			@include icon.generate('arrow_bottom');
```

File: packages/scss/src/components/tableSorted/states.scss [hunk 12]
```diff
@@ -161,7 +161,7 @@
 
 	.table-head-row-cell-sortableButton {
 		&::before {
-			@include icon.generate('arrow_full_north');
+			@include icon.generate('arrow_north');
```
Comment (AlexiaAlarza) [inline packages/scss/src/components/tableSorted/states.scss:164]:
```suggestion
			@include icon.generate('arrow_top');
```

File: stories/documentation/actions/button/button-group.stories.ts [hunk 13]
```diff
@@ -41,7 +41,7 @@ function getTemplate(args: ButtonGroupStory): string {
 		<li class="button-group-item"><button type="button" class="button ${outlined} ${classes}">Bouton</button></li>
 		<li class="button-group-item">
 			<button type="button" class="button ${outlined} ${classes} mod-more">
-				<span class="lucca-icon icon-chevronSouth" aria-hidden="true"></span>
+				<span class="lucca-icon icon-arrowChevronBottom" aria-hidden="true"></span>
```
Comment (AlexiaAlarza) [inline stories/documentation/actions/button/button-group.stories.ts:44]: it's weird, in the figma file this icon name is "icon-arrowChevronDown"

Comment (AlexiaAlarza) [reply stories/documentation/actions/button/button-group.stories.ts:44]: we may need to harmonize between Bottom and Down

File: stories/documentation/forms/files/files-uploading.stories.ts [hunk 14]
```diff
@@ -48,7 +46,7 @@ function getTemplate(args: FilesUploadingStory): string {
 			</span>
 		</span>
 	</label>
-	`
+	`;
```
Comment (AlexiaAlarza) [inline stories/documentation/forms/files/files-uploading.stories.ts:49]: is it an error ?

File: stories/documentation/forms/timepicker/timepicker-basic.stories.ts [hunk 15]
```diff
@@ -34,10 +34,10 @@ function getTemplate(args: timepickerBasicStory): string {
 			<input class="timepicker-field-input" type="number" min="0" max="23" step="1" value="7">
 		</label>
 		<button class="timepicker-field-increment" type="button" tabindex="-1" aria-hidden="true">
-			<span class="lucca-icon icon-northArrow"></span>
+			<span class="lucca-icon icon-chevronTop"></span>
```
Comment (AlexiaAlarza) [inline stories/documentation/forms/timepicker/timepicker-basic.stories.ts:37]: on figma file the name is "icon-chevronUp"

File: stories/documentation/forms/timepicker/timepicker-basic.stories.ts [hunk 16]
```diff
@@ -34,10 +34,10 @@ function getTemplate(args: timepickerBasicStory): string {
...
 		</button>
 		<button class="timepicker-field-increment" type="button" tabindex="-1" aria-hidden="true">
-			<span class="lucca-icon icon-southArrow"></span>
+			<span class="lucca-icon icon-chevronBottom"></span>
```
Comment (AlexiaAlarza) [inline stories/documentation/forms/timepicker/timepicker-basic.stories.ts:40]: icon-chevronDown ?

File: stories/documentation/forms/timepicker/timepicker-basic.stories.ts [hunk 17]
```diff
@@ -47,10 +47,10 @@ function getTemplate(args: timepickerBasicStory): string {
 			<input class="timepicker-field-input" type="number" min="00" max="45" step="15" value="00">
 		</label>
 		<button class="timepicker-field-increment" type="button" tabindex="-1" aria-hidden="true">
-			<span class="lucca-icon icon-northArrow"></span>
+			<span class="lucca-icon icon-chevronTop"></span>
```
Comment (AlexiaAlarza) [inline stories/documentation/forms/timepicker/timepicker-basic.stories.ts:50]: idem

File: stories/documentation/forms/timepicker/timepicker-basic.stories.ts [hunk 18]
```diff
@@ -47,10 +47,10 @@ function getTemplate(args: timepickerBasicStory): string {
...
 		</button>
 		<button class="timepicker-field-increment" type="button" tabindex="-1" aria-hidden="true">
-			<span class="lucca-icon icon-southArrow"></span>
+			<span class="lucca-icon icon-chevronBottom"></span>
```
Comment (AlexiaAlarza) [inline stories/documentation/forms/timepicker/timepicker-basic.stories.ts:53]: idem

File: stories/documentation/forms/files/files-uploading.stories.ts (hunk 14 above)
Comment (vvalentin-lucca) [reply stories/documentation/forms/files/files-uploading.stories.ts:49]: It must be the linter.

File: stories/documentation/actions/button/button-group.stories.ts (hunk 13 above)
Comment (vvalentin-lucca) [reply stories/documentation/actions/button/button-group.stories.ts:44]: On the code side, everything is top/bottom. I go back to Clément, well done!

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2170/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/5727/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2170/storybook

File: stories/documentation/overlays/dropdown/dropdown-directive.stories.ts [hunk 19]
```diff
@@ -17,19 +17,19 @@ const Template: Story = (args) => ({
...
 		<a routerLink="." fragment="link2" class="lu-dropdown-options-item-action" luDropdownItem>
-			<span aria-hidden="true" class="lucca-icon icon-edit"></span>
+			<span aria-hidden="true" class="lucca-icon icon-officePen"></span>
 			Editer
 		</a>
 	</li>
 	<li class="lu-dropdown-options-item">
 		<button type="button" class="lu-dropdown-options-item-action mod-delete" luDropdownItem>
-			<span aria-hidden="true" class="lucca-icon icon-trash"></span>
+			<span aria-hidden="true" class="lucca-icon icon-trashDelete"></span>
```
Comment (jeremie-lucca) [inline stories/documentation/overlays/dropdown/dropdown-directive.stories.ts:32]: Trash + Delete might be too much ?

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2170/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/5796/ui-diff)

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/5797/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2170/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/5801/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2170/storybook

**Review by jeremie-lucca — DISMISSED** [review]

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/5802/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2170/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/5805/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2170/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/5836/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2170/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/5839/ui-diff)

**Comment by LuccaIntegration** [comment]
:woman_cook: http://lucca-front.lucca.local/PR-2170/storybook

**Comment by CCNET-iLucca** [comment]
[Tests d'interfaces](https://qa.lucca.tech/5840/ui-diff)

**Review by jeremie-lucca — APPROVED** [review]

**Review by GuillaumeNury — APPROVED** [review]

**Pull request merged**